doc_source/private-clusters.md

```diff
@@ -0,0 +1,80 @@
++ Autoscaling requires the Autoscaling VPC endpoint\.
++ [App Mesh](appmesh-getting-started) is supported with private clusters when using the App Mesh Envoy VPC endpoint\.
+  + The App Mesh sidecar injector for Kubernetes is supported\.
+  + The App Mesh controller for Kubernetes is not supported\.
+
+## Creating local copies of container images<a name="creating-local-copies-of-container-images"></a>
+
+Since a private cluster has no outbound internet access, container images cannot be pulled from external sources, such as Docker Hub. Instead, container images must be copied locally to Amazon ECR, or alternatively a registry running in the VPC.
+
+A container image can be copied to Amazon ECR from outside the private VPC. The private cluster will access the ECR repository using the ECR VPC endpoints. You must have Docker and the AWS CLI installed on the workstation used to create the local copy.
+
+**To create a local copy of a container image** *{SOURCE_IMAGE}*
+
+1. [Create a repository](https://docs.aws.amazon.com/AmazonECR/latest/userguide/repository-create.html)\.
+
+1. Pull the container image from the external registry using **docker pull**\.
+
+1. Tag your image with the Amazon ECR registry, repository, and optional image tag name combination using **docker tag**\.
+
+1. [Authenticate to the registry](https://docs.aws.amazon.com/AmazonECR/latest/userguide/Registries.html#registry_auth)\.
+
+1. [Push the image to Amazon ECR](https://docs.aws.amazon.com/AmazonECR/latest/userguide/docker-push-ecr-image.html)\ using **docker push**.
+
+**Note**
+Be sure to update your resource configuration to use the new image location.
+
+The following example pulls the [amazon/aws-node-termination-handler](https://hub.docker.com/r/amazon/aws-node-termination-handler) image (using tag **v1.3.1-linux-amd64**) from Docker Hub and creates a local copy in Amazon ECR.
+
+```
+aws ecr create-repository --repository-name amazon/aws-node-termination-handler
+docker pull amazon/aws-node-termination-handler:v1.3.1-linux-amd64
+docker tag amazon/aws-node-termination-handler *aws_account_id*.dkr.ecr.*region*.amazonaws.com/amazon/aws-node-termination-handler:v1.3.1-linux-amd64
+aws ecr get-login-password --region *REGION* | docker login --username AWS --password-stdin *aws_account_id*.dkr.ecr.*region*.amazonaws.com
+docker push *aws_account_id*.dkr.ecr.*region*.amazonaws.com/amazon/aws-node-termination-handler:v1.3.1-linux-amd64
+```
+
+## VPC endpoints for private clusters<a name="vpc-endpoints-for-private-clusters"></a>
+
+The following [VPC endpoints](https://docs.aws.amazon.com/vpc/latest/userguide/vpc-endpoints.html) may be required.
+
++ **com.amazonaws.REGION.ec2**
++ **com.amazonaws.REGION.ecr.api**
++ **com.amazonaws.REGION.ecr.dkr**
++ **com.amazonaws.REGION.s3** for pulling container images
++ **com.amazonaws.REGION.logs**
```

Comment:
Basically, for launching fargate pods, endpoints for ecr, s3 and sts are required to have. All other endpoints are optional and depend on your needs.